runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1260,6 +1260,29 @@ gcParseXXArguments(J9JavaVM *vm)
 		}
 	}
 
+#if defined(J9VM_ARCH_X86)
+	{
+		if (NULL != vm->sharedClassPreinitConfig) {
+			IDATA argIndexXaot = FIND_ARG_IN_VMARGS(OPTIONAL_LIST_MATCH, "-Xaot", 0);
+			IDATA argIndexXnoaot = FIND_ARG_IN_VMARGS(OPTIONAL_LIST_MATCH, "-Xnoaot", 0);
+			if (argIndexXaot >= argIndexXnoaot) { 
+				IDATA portableSharedCacheIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXPORTABLESHAREDCACHE, NULL);
+				IDATA noportableSharedCacheIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOPORTABLESHAREDCACHE, NULL);
+				if (portableSharedCacheIndex != noportableSharedCacheIndex) {
+					/* At least one option is set. Find the right most one. */
+					if (portableSharedCacheIndex > noportableSharedCacheIndex) {
+						extensions->portableSharedCacheEnabled = true;
+					} else {
+						extensions->portableSharedCacheEnabled = false;
+					}
+				} else {
+					extensions->shouldForceCeilingShiftForContainers = true;
+				}
+			}
+		}
+	}
+#endif /* defined(J9VM_ARCH_X86) */
+
```

Comment:
There are a few things I have problem with:
- If we want to enable feature for x86 platforms only we should have boolean responsible to enable/disable feature.
An initialization of this variable should be only place where `#if defined(J9VM_ARCH_X86)` is used. All other related code should be platform independent (should exist for all platforms and not for x86 only. This helps in future to expand feature to other platforms.
- There is a parsing of options parsed earlier. Would be better to reuse result of previous parsing (and maybe even deeper - results of logical analysis made based on previous parsing)
- Parsing of options under logical condition might lead to potential existence of cases with unconsumed arguments. Parsing of provided options should be done regardless of applicability of them. I understand this is second parsing, so see previous bullet
- Last but not least this is absolutely wrong place to set `extensions->shouldForceCeilingShiftForContainers` in case if none of the options are discovered. This new value for variable must be initialized as default before this parsing in a proper place. 

I believe this is wrong place for this code. I suggest:
- create required booleans somewhere in VM substructure (find a right place)
- initialize them properly at the time of first options parsing in `jvminit.c`
- if it is possible to add such booleans visible from OMR code you don't need these booleans in GCExtensionsBase. Use OMR VM variables instead directly.
- if such booleans are visible from OpenJ9 only you still need new variables in GCExtensionsBase. These variables should be initialized properly for standalone OMR and be set at OpenJ9 side accordinly (somewhere in `mminit.cpp` I guess)

So I believe we don't need any change in this file at all. This is wrong place to do it. There is a much better way to do it.